### DISCUSSION TITLE
Actor scheduler

### DISCUSSION THREAD

**Discussion by EnvOut** [slawlor/ractor#81] [Q&A] [answered]
I need to change the value of every tick. Are there any actors that implement this approach, or do I need to use the scheduler separately and pass the message to the actor?

**Comment by slawlor** [comment, marked as answer]
So you could set it up relatively basically with `time::send_after` which will just send a single message after a delay. It's equivalent of like "sleeping" in a loop without blocking the actor however. So something like

```rust
impl Actor for DynamicTimeActor {
  type Msg = Duration;
  type State = ();
  type Arguments = Duration;

  async fn pre_start(&self, myself: ActorRef<Self>, startup_delay: Self::Arguments) -> Result<Self::State, ActorProcessingErr> {
    myself.send_after(startup_delay, || Duration::from_millis(random_time()));
    Ok(())
  }

  async fn handle(&self, myself: ActorRef<Self>, message: Self::Msg, _: &mut Self::State) -> Result<(), ActorProcessingErr> {
    myself.send_after(message, || Duration::from_millis(random_time()));
    Ok(())
  }

}
```

This isn't a functional example, but hopefully it demonstrates the idea
